src/freenet/client/filter/TheoraPacketFilter.java

```diff
@@ -53,37 +55,37 @@ public CodecPacket parse(CodecPacket packet) throws IOException {
     private void verifyIdentificationHeader(BitInputStream input) throws IOException {
         verifyTypeAndHeader("Identification", input, 0x80); // expected -128
 
-        checkHeaderField("Identification", "VMAJ", input, 8, v -> v != 3);
+        checkHeaderField("Identification", "VMAJ", input, 8, not(v -> v != 3));
```

Comment:
The usage of `not` makes these harder to read than if they had been inverted, I think.

Reply:
On this line I selected this style because it is closer to spec, I think. Spec says:

> Read an 8-bit unsigned integer as VMAJ.  If VMAJ is not 3, then stop.This stream is not decodable according to this specification.

So, '**not 3**' and `v -> v != 3` looks closer then `v -> v == 3`. So can I please known your opinion one more time?
I will recheck other lines and think some `not`-s should be removed.

Reply:
Mostly my argument is that using the `not` function on all of the expressions does make some of them harder to read. Sure, for `not(v -> v != 3)` that is easy but an expression like (the completely made up) `not(v -> v != 3 && v != 4 || v != 17)` I have a tough time understanding which values are actually valid.

Reply:
I also removed `VREV == 1` check, as:

> Read an 8-bit unsigned integer as VREV.  If VREV is greater than 1, then this stream may contain optional features or interpretational changes documented  in  a  future  version  of  this  specification.   **Regardless**  of  the value of VREV, the stream is decodable according to this specification.